tty/uart16550/uarthw-zynq7000.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Phoenix-RTOS
+ *
+ * UART 16550 device driver
+ *
+ * Hardware abstraction layer (armv7a9-zynq7000)
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <board_config.h>
+
+#include "uart16550.h"
+
+#ifndef UART16550_BASE0
+#define UART16550_BASE0 0x43c10000
+#endif
+
+#ifndef UART16550_IRQ0
+#define UART16550_IRQ0 61
+#endif
+
+#ifndef UART16550_BASE1
+#define UART16550_BASE1 0x43c20000
+#endif
+
+#ifndef UART16550_IRQ1
+#define UART16550_IRQ1 62
+#endif
+
+#ifndef UART16550_BASE2
+#define UART16550_BASE2 0x43c30000
+#endif
+
+#ifndef UART16550_IRQ2
+#define UART16550_IRQ2 63
+#endif
+
+#ifndef UART16550_BASE3
+#define UART16550_BASE3 0
+#endif
+
+#ifndef UART16550_IRQ3
+#define UART16550_IRQ3 0
+#endif
+
+
+typedef struct {
+	volatile uint32_t *base;
+	unsigned int irq;
+} uarthw_ctx_t;
+
+
+uint8_t uarthw_read(void *hwctx, unsigned int reg)
+{
+	volatile uint32_t *p = (((uarthw_ctx_t *)hwctx)->base + reg);
+
+	return ((uint8_t)(*p));
+}
+
+
+void uarthw_write(void *hwctx, unsigned int reg, uint8_t val)
+{
+	volatile uint32_t *p = (((uarthw_ctx_t *)hwctx)->base + reg);
+
+	*p = val;
+
+	return;
+}
+
+
+char *uarthw_dump(void *hwctx, char *s, size_t sz)
+{
+	snprintf(s, sz, "base=0x%p irq=%d", (volatile void *)(((uarthw_ctx_t *)hwctx)->base), ((uarthw_ctx_t *)hwctx)->irq);
```

Comment:
`irq` is unsigned, so maybe better is `%u` (for ia32 and riscv too).

Reply:
Done